qiskit_machine_learning/algorithms/classifiers/pegasos_qsvc.py

```diff
@@ -14,15 +14,19 @@
 
 import logging
 from datetime import datetime
-from typing import Optional, Dict
+from typing import Optional, Dict, Union
 
 import numpy as np
 from qiskit.utils import algorithm_globals
 from sklearn.base import ClassifierMixin
 
 from ...algorithms.serializable_model import SerializableModelMixin
 from ...exceptions import QiskitMachineLearningError
-from ...kernels.quantum_kernel import QuantumKernel
+from ...kernels import BaseKernel, FidelityQuantumKernel
+from ...kernels.quantum_kernel import QuantumKernel as QuantumKernelOld
+
+QuantumKernel = Union[QuantumKernelOld, BaseKernel]
```

Comment:
I'd prefer to keep `QuantumKernel` separated from the new hierarchy. Especially considering it's training capabilities. While inheriting from `BaseKernel` may be not so bad, but adding the mixin will make it over complicated, the ways how kernel elements are computed are too different now. Since we want to deprecate the existing one, I don't really want to change it now.

Reply:
All right, now we have 
- `BaseKernel`
- `FidelityQuantumKernel(BaseKernel)`
- `TrainableKernel(BaseKernel, ABC)`
- `FidelityTrainableQuantumKernel(TrainableKernel, FidelityQuantumKernel)`
- `QuantumKernel(BaseKernel, TrainableKernel)`
Additional tweaks of `**kwargs` may be required in the future to support more inheritance patterns.